x-pack/plugins/siem/server/lib/timeline/routes/draft_clean_timelines_route.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from '../../../../../../../src/core/server';
+import { ConfigType } from '../../..';
+import { transformError, buildSiemResponse } from '../../detection_engine/routes/utils';
+import { TIMELINE_DRAFT_CLEAN_URL } from '../../../../common/constants';
+import { buildFrameworkRequest } from './utils/common';
+import { SetupPlugins } from '../../../plugin';
+import { getDraftTimeline, resetTimeline, getTimeline, persistTimeline } from '../saved_object';
+import { draftTimelineDefaults } from '../default_timeline';
+
+export const draftCleanTimelinesRoute = (
+  router: IRouter,
+  config: ConfigType,
+  security: SetupPlugins['security']
+) => {
+  router.post(
+    {
+      path: TIMELINE_DRAFT_CLEAN_URL,
+      validate: {},
+      options: {
+        tags: ['access:siem'],
+      },
+    },
+    async (context, request, response) => {
+      const frameworkRequest = await buildFrameworkRequest(context, security, request);
+      const siemResponse = buildSiemResponse(response);
+
+      try {
+        const {
+          timeline: [draftTimeline],
+        } = await getDraftTimeline(frameworkRequest);
+
+        if (draftTimeline?.savedObjectId) {
+          await resetTimeline(frameworkRequest, [draftTimeline.savedObjectId]);
```

Comment:
I am wondering why the `resetTimeline` cannot return the new timeline right away.